sdklab/fuzzing/fuzz_send_message.py

```diff
@@ -0,0 +1,118 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import asyncio
+import logging
+import argparse
+from azure.iot.device.aio import IoTHubDeviceClient
+from test_utils import test_env, paho_fuzz_hook, random_content
+from test_utils.service_helper import ServiceHelper
+
+logging.basicConfig(level=logging.WARNING)
+logging.getLogger("e2e").setLevel(level=logging.INFO)
+
+
+async def queue_send_messages(device_client, messages_to_send=60):
+    messages = [random_content.get_random_message() for _ in range(0, messages_to_send)]
+    message_ids = [m.message_id for m in messages]
+
+    send_tasks = [asyncio.create_task(device_client.send_message(m)) for m in messages]
+
+    return (send_tasks, message_ids)
+
+
+async def wait_for_messages(service_helper, message_ids):
+    count_received = 0
+    while len(message_ids):
+        message = await service_helper.wait_for_eventhub_arrival(message_id=None)
+        if message.message_id in message_ids:
+            message_ids.remove(message.message_id)
+        count_received += 1
+        print("received={}, remaining={}".format(count_received, len(message_ids)))
+
+
+drop_outgoing_packets_until_reconnect = 1
+drop_individual_outgoing_packets = 2
+drop_incoming_packets_until_reconnect = 3
+drop_individual_incoming_packets = 4
+raise_send_exception = 5
+raise_receive_exception = 6
+
+fuzz_type_help = """
+drop_outgoing_packets_until_reconnect = 1
+drop_individual_outgoing_packets = 2
+drop_incoming_packets_until_reconnect = 3
+drop_individual_incoming_packets = 4
+raise_send_exception = 5
+raise_receive_exception = 6
+"""
+
+
+async def main(fuzz_type):
+    device_client = IoTHubDeviceClient.create_from_connection_string(
+        test_env.DEVICE_CONNECTION_STRING, keep_alive=15
+    )
+    service_helper = ServiceHelper(
+        iothub_connection_string=test_env.IOTHUB_CONNECTION_STRING,
+        eventhub_connection_string=test_env.EVENTHUB_CONNECTION_STRING,
+        eventhub_consumer_group=test_env.EVENTHUB_CONSUMER_GROUP,
+    )
+
+    paho_fuzz_hook.add_paho_logging_hook(device_client)
+
+    try:
+
+        # Connect the device client.
+        print("connecting")
+        await device_client.connect()
+
+        # Start fuzzing after the client is connected
```

Comment:
I feel as though this module needs a bit more documentation explaining what's going on here, what's being tested, and why it matters